### ISSUE TITLE
Help: chatGuid + and - meaning

### ISSUE THREAD

**Issue by User8395** [BlueBubblesApp/bluebubbles-server#681]
When sending a message via the REST API, the `chatGuid` is as follows
```
"chatGuid": "iMessage;-;email@email.com"
```

My question is why is there a minus in the middle and sometimes a plus, and why is the "Chat does not exist" error thrown when using a plus instead of a minus?

**Comment by mehapps** [comment]
So i dont think I've personally seen a + before in the chatGuid, would u happen to have an example of this? (not a bb dev though so I have WAY WAY less knowledge abt how imessage works)

**Comment by User8395** [comment]
> So i dont think I've personally seen a + before in the chatGuid, would u happen to have an example of this? (not a bb dev though so I have WAY WAY less knowledge abt how imessage works)

![20240817_130217.jpg](https://github.com/user-attachments/assets/2a66fa27-cc6b-440b-b1f5-0a12e0d07dd0)

**Comment by zlshames** [comment]
DMs will be iMessage;-;<address> while group chats are iMessage;+;chat<number>

**Comment by User8395** [comment]
> DMs will be iMessage;-;{address} while group chats are iMessage;+;chat{number}

Can you give an example of sending to a group with 1234567890, bob@bob.com, and 6666666666?

**Comment by mehapps** [comment]
I guess then you would get the chatguid from that group (I usually get it through a new-message webphook) and then use it

**Comment by User8395** [comment]
> I guess then you would get the chatguid from that group (I usually get it through a new-message webphook) and then use it

How would that work? Does it show the guid when listing chats?

**Comment by zlshames** [comment]
Yeah get the chat guid from an API call or something. If you have a group chat name, it'll be easily identifiable from the chat endpoint

**Comment by mehapps** [comment]
> > I guess then you would get the chatguid from that group (I usually get it through a new-message webphook) and then use it
> 
> How would that work? Does it show the guid when listing chats?

I setup the new-message webhook and then an HTTP server (there should be an example in python somewhere) then I would wait for a message from that group chat. Once it arrives look at the logs and see what the chatGUID is.
There's probably a different way to do it (maybe you could send a message to the group chat and use message updates
